### ISSUE TITLE
flameshot gui doesn't start, no errors

### ISSUE THREAD

**Issue by InconsolableCellist** [flameshot-org/flameshot#1648]
Reopened. See #90

```
$ flameshot --version
Flameshot v0.9.0
Compiled with Qt 5.11.3
```

Running `flameshot` starts the system tray icon 

Running `flameshot gui` immediately exits, exit code `0`

```
$ cat /etc/os-release 
PRETTY_NAME="Devuan GNU/Linux 3 (beowulf)"
NAME="Devuan GNU/Linux"
VERSION_ID="3"
VERSION="3 (beowulf)"
VERSION_CODENAME=beowulf
ID=devuan
ID_LIKE=debian
HOME_URL="https://www.devuan.org/"
SUPPORT_URL="https://devuan.org/os/community"
BUG_REPORT_URL="https://bugs.devuan.org/"
```

```
$ uname -r
4.19.0-14-amd64
```

Running `openbox`

Also did `killall -9 flameshot` then `flameshot &` and `flameshot gui` with the same behavior

This distro and window manager worked before, but I did a reinstall and something's broken, possibly due to a newer version of flameshot

Installed with `flameshot-0.9.0-1.debian-10.amd64.deb` from https://github.com/flameshot-org/flameshot/releases/tag/v0.9.0

**Comment by mmahmoudian** [comment]
Unfortunately some users in #90 are not actins as grownups and happy with downvoting without having the curtesy of providing valid input that can be used for bug tracking and reproduction.

Regarding your issue though, I'm sorry, but I failed to find  the info regarding your window manager. If you are using a window manager with that doesn't have systray or notification manager, you might get this behaviour due to lack of these. Would you please make sure you have both on your system and reply here?

**Comment by InconsolableCellist** [comment]
My window manager is `openbox`, and I'm running `xfce4-panel` which has a notification area (and the icon appears there)

It appears after `flameshot &` in a terminal, but nothing happens with `flameshot gui` other than it immediately returns with an exit code of 0.


![screenshot-2021_05_29-032833_PM](https://user-images.githubusercontent.com/23345188/120085101-91466500-c092-11eb-9c22-e41fe57c8fb9.jpg)


Also it looks like the first time I run `flameshot gui` focus is changed from the terminal to some invisible window, which doesn't happen on subsequent runs without doing `killall flameshot`

**Comment by mmahmoudian** [comment]
@InconsolableCellist thank you for the quick response 👍🏼

I wonder if this behaviour has anything to do with your monitors and graphics card. Is it the same as before? Also if you gave multiple monitors, what is the configuration and resolutions. Additionally, are the monitors connected to the same graphics card?

**Comment by InconsolableCellist** [comment]
It could have something to do with that, but that configuration hasn't changed. I'm on a GTX 970  running driver 390.141 (which is a bit old), with two monitors I've used before. However, that got me thinking: I used xrandr to disable my non-primary display and the flameshot gui actually popped up, though I couldn't complete the selection and had to kill it via a tty. I then killed flameshot and started it again in the same manner as above, and this time it's working. I'll re-enable my other monitor and see what happens.

Could this be caused by a configuration change to the monitors, or maybe one of them going to sleep or having its input switched?

**Comment by mmahmoudian** [comment]
Thanks for investigating that part. I don't know what is cause yet, but that was my hypothesis. Unfortunately this type of issues are not reproducible on VM as it might have something todo with hardware and drivers and etc.

Perhaps @borgmanJeremy or @ZetaoYang might have some idea on what is happening or how we can narrow down the search further.

**Comment by InconsolableCellist** [comment]
After running xrandr again and re-enabling both monitors, it seems to show up as expected now

**Comment by mmahmoudian** [comment]
So if it was setup with two monitors and one of them goes away Flameshot doesn't open? And now that you have both monitors enabled Flameshot is working normally?

**Comment by InconsolableCellist** [comment]
That's right, and I'm able to reproduce it pretty simply now:

1. Have flameshot running
2. Have a configuration for two monitors
```
DP-0 connected primary 3840x1600+3840+0 (normal left inverted right x axis y axis) 879mm x 366mm
   3840x1600     59.99*+  75.00    29.99  
DP-4 connected 3840x2160+0+0 (normal left inverted right x axis y axis) 530mm x 300mm
   3840x2160     60.00*+  60.00    59.94  
```
3. Turn off the left monitor (DP-4)
4. Run `flameshot gui`

Observed: 
Mouse focus is taken away but nothing else happens. Upon turning on the left monitor it displays the contents of the right monitor with the grey flameshot gui overlay on top of it, but it cannot be interacted with.

**Comment by mmahmoudian** [comment]
Ok, this clears many things. Thanks for putting time into this. I think the most important step is your step 1 that the Flameshot should be running when one monitor is getting disabled.

Let's see if @borgmanJeremy or @thepurple have ideas on how to address this.

**Comment by InconsolableCellist** [comment]
Actually I just retested and was mistaken. Step 1 can be:

1. Have flameshot running OR confirm that flameshot is not running

I did `killall flameshot` and then turned off the monitor (but left the monitor configuration the same) and repeated the steps, and saw the same behavior. It's like it's confused by the missing monitor and won't do it on the still-connected one.

**Comment by ZetaoYang** [comment]
> 
> 
> Reopened. See #90
> 
> ```
> $ flameshot --version
> Flameshot v0.9.0
> Compiled with Qt 5.11.3
> ```
> 
> Running `flameshot` starts the system tray icon
> 
> Running `flameshot gui` immediately exits, exit code `0`
> 
> ```
> $ cat /etc/os-release 
> PRETTY_NAME="Devuan GNU/Linux 3 (beowulf)"
> NAME="Devuan GNU/Linux"
> VERSION_ID="3"
> VERSION="3 (beowulf)"
> VERSION_CODENAME=beowulf
> ID=devuan
> ID_LIKE=debian
> HOME_URL="https://www.devuan.org/"
> SUPPORT_URL="https://devuan.org/os/community"
> BUG_REPORT_URL="https://bugs.devuan.org/"
> ```
> 
> ```
> $ uname -r
> 4.19.0-14-amd64
> ```
> 
> Running `openbox`
> 
> Also did `killall -9 flameshot` then `flameshot &` and `flameshot gui` with the same behavior
> 
> This distro and window manager worked before, but I did a reinstall and something's broken, possibly due to a newer version of flameshot
> 
> Installed with `flameshot-0.9.0-1.debian-10.amd64.deb` from https://github.com/flameshot-org/flameshot/releases/tag/v0.9.0

Can you provide us with the flameshot print log?

**Comment by zachliu** [comment]
interesting, i had a somewhat similar issue
i've been using
```
$ flameshot --version
Flameshot v0.9.0
Compiled with Qt 5.12.8
```
until i upgraded my system from `Linux Mint 20.1` to `20.2`
then my flameshot cannot start
Running flameshot ===> NO system tray icon
Running flameshot gui ===> immediately exits, exit code 0 (i can see the gui for a split second)

so i just pull down that latest `v0.10.0`, `cmake` and then `make` now i have
```
$ flameshot --version
Flameshot v0.10.0
Compiled with Qt 5.12.8
```
everything works again

**Comment by Aditeya** [comment]
I just ran into the same issue
I'm running Manjaro linux + bspwm + polybar

```
[aditeya@nemo ~]$ flameshot --version
Flameshot v0.9.0
Compiled with Qt 5.15.2
```
I changed the config to hide the systray icon and thats when it stopped working after a reboot. After changing `disabledTrayIcon` back to false in the config it started to work as normal again.

**Comment by mmahmoudian** [comment]
@Aditeya I think you issue is more close to #1730 than this thread

**Comment by Aditeya** [comment]
I'm sure its this issue because when i try to start the daemon with `disabledTrayIcon` to true, it gives this error:
```
[aditeya@nemo ~]$ flameshot
QSettings::value: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
Segmentation fault (core dumped)
```

**Comment by mmahmoudian** [comment]
@Aditeya  [It has been confirmed](https://github.com/flameshot-org/flameshot/issues/1730#issuecomment-884050509) by another kind user (@chuahou) that the segfault happens for v0.9.0 and has been fixed in v0.10.0.

**Comment by patrickhener** [comment]
```bash
> flameshot
QSettings::value: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
[1]    25192 segmentation fault (core dumped)  flameshot
> flameshot --version
Flameshot v0.10.0
Compiled with Qt 5.15.2
> uname -a
Linux redkite 5.13.4-arch1-1 #1 SMP PREEMPT Tue, 20 Jul 2021 16:58:51 +0000 x86_64 GNU/Linux
```

It might not, though.

**Comment by chuahou** [comment]
> It might not, though.

@patrickhener This bug was fixed in #1742 and this fix is in v0.10.1. The reason why it happened for v0.9.0 and not v0.10.0 previously is that it occurred if there was a newer version of Flameshot available than the one installed, so at the time it did not occur in v0.10.0 since there was no newer version. Now, v0.10.1 has been released so the bug now occurs in v0.10.0 as well.

After #1742 this bug should no longer occur even when there is a newer version.

**Comment by HW71** [comment]
Hi all,

I see similar issue when running Flameshot on my Linux Mint Debian Edition with Cinnamon desktop: after login to Cinnamon no tray-icon shows up - but using "ps" I see that a process is running. Here are some details:

```
flameshot --version
Flameshot v0.10.1
Compiled with Qt 5.11.3
```

```
uname -r
4.19.0-17-amd64
```

Starter / .desktop file for Flameshot:
```
Exec=/usr/bin/flameshot
Icon=org.flameshot.Flameshot
Terminal=false
Type=Application
Categories=Graphics;
StartupNotify=false
Actions=Configure;Capture;Launcher;
X-DBUS-StartupType=Unique
X-DBUS-ServiceName=org.flameshot.Flameshot
X-KDE-DBUS-Restricted-Interfaces=org.kde.kwin.Screenshot
X-GNOME-Autostart-enabled=true
```

Process / PID running after login (but without showing tray icon):
```
erik@thinkpad:~$ ps -aef |grep flame
erik      8686  8300  0 21:27 ?        00:00:00 /usr/bin/flameshot
erik     10417 10414  0 21:28 pts/0    00:00:00 grep flame
```
If I kill that one and restart Flameshot using the related "starter" entry from the Cinnamon menu the tray icon shows-up as expected.

Do you have any suggestions on how to track this issue down? Or is this some kind of usage-error?

Thanks so much in advance & kind regards!

Cheers, Erik

**Comment by mmahmoudian** [comment]
@HW71 what do you get (output and systray) when you do:
1. Kill all instances of Flameshot `pkill flameshot`
2. Run Flameshot `flameshot`

-------
Also as a side tip, you can use `pgrep flameshot` to get the PID.

**Comment by HW71** [comment]
Hi @mmahmoudian,

thanks for coming back to this. If I kill the instance and manually start flameshot from the command line, then

1. the icon shows up in the systray (that's the same result as if I manually start the application from the Cinnamon Start menu)
2. I get the following console output:
```
erik@thinkpad:~$ flameshot
qt5ct: using qt5ct plugin
QSettings::value: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
qt5ct: D-Bus system tray: yes
qt5ct: custom style sheet is disabled
```

Is this what you were after? Thanks again & keeping fingers crossed we get this solved somehow!

Kind regards, Erik

**Comment by mmahmoudian** [comment]
@HW71 yes this is exactly what I was after. This does that Flameshot works as expected. Note I need three more information:
1. The result of `which flameshot`
2. Did I understood correctly that when you log into your Cinnammon Flameshot automatically starts but it lacks the tray icon?
3. If the answer to the point above is TRUE, then as I'm not familiar with the Cinnamon, please figure out how Cinnamon runs applications at startup. Basically what binary does it run. Does it match with the result of `which flameshot`?

**Comment by HW71** [comment]
Hi there @mmahmoudian,

awesome - thanks a lot for the confirmation and your list of additional things to check. As I did some more tests yesterday I noticed, that more tray icons (of other applications) were missing. Hence I was quite sure that this is **not** an issue with Flameshot. 

Now in the meantime I was able to fix the problem => the trick was simply to configure a delay for the autostart of all applications where the tray icon was missing. Now the icons show up again after login (including the one of Flameshot => Hurray! :-)  ).

My best guess on this: it looks like a race condition to me, but in any case this looks more like a Cinnamon issue rather than a problem of Flameshot => if a process gets started by the "autostart" functionality then also the related tray icon should be displayed. Maybe I will check back with the Cinnamon guys on this.

So just in case others face a similar issue: I changed the ".desktop" launcher located in the ~/.config/autostart directory by adding the line displayed in bold at the end of the "[Desktop Entry]" section:

> **[Desktop Entry]**
> Name=Flameshot
> ...
> Comment=Powerful yet simple to use screenshot software.
> ...
> Keywords=flameshot;screenshot;capture;shutter;
> ...
> Exec=/usr/bin/flameshot
> Icon=org.flameshot.Flameshot
> ...
> X-DBUS-ServiceName=org.flameshot.Flameshot
> X-KDE-DBUS-Restricted-Interfaces=org.kde.kwin.Screenshot
> **X-GNOME-Autostart-Delay=3**
> ...

Not quite sure, if a delay of 1 second would do the trick, too - I didn't check that yet...

So thanks again for taking care, @mmahmoudian and have a great one!

Cheers, Erik

**Comment by mmahmoudian** [comment]
@HW71 Thanks for the confirmation and following up the issue. Also sorry for the typos in my previous command. I typed it on my phone and the autocorrect did some automistakes 😅 Now the typos (those which I spotted) are fixed.

**Comment by HW71** [comment]
> 
> 
> @HW71 Thanks for the confirmation and following up the issue. Also sorry for the typos in my previous command. I typed it on my phone and the autocorrect did some automistakes 😅 Now the typos (those which I spotted) are fixed.

No problem - looks like my "AutoCorrection" while reading worked well as I didn't notice any typos at all!  ;-) Thanks also for adding a hint to the FAQ => hopefully this helps others facing a similar issue... :-)

Cheers, Erik
